src/elements/types/ObjectProperty.js

```diff
@@ -53,6 +54,8 @@ function readKey(children) {
         children.isNode('Identifier')
     ) {
         return children.passNode();
+    } else if (children.isToken('Identifier')) {
+        return new Identifier([children.passToken()]);
```

Comment:
So babylon has a design flaw, when you have an `ObjectProperty` and `AssignmentPattern` inside, it will duplicate the `Identifier` node in two places – https://github.com/babel/babylon/issues/49, which means that during traversing you have guard protecting you against second run, i.e.:

```js
let i = 0;

let ast = require("babylon").parse("({ test = 1 } = {})", {});

let result = require('estraverse').traverse( ast.program, {
    keys: require("babel-types").VISITOR_KEYS,
    enter: function (node) {
        if (node.type === 'Identifier') {
            i++;
        }
    }
});

console.log(i); // 2
```
Or if you manipulating you will have to duplicate your the efforts. In order to mitigate that we remove the duplicate `Identifier` node from the `AssignmentPattern` in `elementTree` module.

Which is why you have to recreate it here, there is probably a simpler but it requires some tinkering